spec/unit/puppetfile_spec.rb

```diff
@@ -127,6 +127,18 @@
 
       expect { subject.add_module('puppet/test_module', module_opts) }.to raise_error(R10K::Error, /cannot manage content.*is not within/i).and not_change { subject.modules }
     end
+
+    it "should disable and not add modules that conflict with the environment" do
+      env = instance_double('R10K::Environment::Base')
+      mod = instance_double('R10K::Module::Base', name: 'conflict', origin: :puppetfile)
+      allow(mod).to receive(:origin=).and_return(nil)
+      allow(subject).to receive(:environment).and_return(env)
+      allow(env).to receive(:'module_conflicts?').with(mod).and_return(true)
+
+      allow(R10K::Module).to receive(:new).with('test', anything, anything, anything).and_return(mod)
+      expect { subject.add_module('test', {}) }.not_to change { subject.modules }
+      expect(subject.disabled_modules).to eq [mod]
```

Comment:
It looks like this is the only place `disabled_modules` is used. Could we maybe check that the `mod` _isn't_ included in `subject.modules`, rather than adding a whole new attr just for testing?

Reply:
The line above this should do that, yes?

I originally added `@disabled_modules` because I thought it might be needed to log or debug something somewhere. However, I never ended up doing that. We could fully remove it right now with no effect. Should I just do that? Seems like probably...

Reply:
I think so, yeah. There's already enough confusing stuff in this repo, so less is more imo.

I guess the first line kind of does that? so I don't feel too strongly about changing that part. it feels a _little_ more implementation-aware than seems ideal, but I think that's a larger problem in this repo 😅